### PR TITLE
Fix rubocop rails errors

### PR DESCRIPTION
**Pull request by macanudo527**
#### What? Why?

- Helps #6055 

This fixes 4 Rubocop Rails errors that can not be safely autocorrected.  

#### What should we test?
Since no new features were added, manual testing shouldn't be needed.

#### Release notes

Changelog Category: Technical changes

### CODE REVIEW

File: app/models/spree/product.rb [hunk 1]
```diff
@@ -257,15 +259,14 @@ def import_date
...
+    def destroy_related_exchange_variants
       transaction do
+        # binding.pry()
         touch_distributors
 
         ExchangeVariant.
           where('exchange_variants.variant_id IN (?)', variants.with_deleted.
           select(:id)).destroy_all
-
-        super
```
Comment (Matt-Yorkley) [inline app/models/spree/product.rb:268]: I'm not sure about this change, it's not doing the same thing as before. Specifically; there's an explicit `transaction` block which was wrapping the whole change, but moving it to a before callback will have different behaviour.

Comment (macanudo527) [reply app/models/spree/product.rb:268]: Should we just ignore this one then? There isn't a way to preserve the `transaction` block and use a callback I think.

Comment (mkllnk) [reply app/models/spree/product.rb:268]: Maybe take the commit out of this PR and work on it in a new PR. You can use an `around_destroy` callback to achieve the same logic. Example:
https://github.com/openfoodfoundation/openfoodnetwork/blob/27dcad205e2b0a2aa3dceb608b11ac849c0eadb3/app/models/spree/variant.rb#L240-L243

File: engines/dfc_provider/app/controllers/dfc_provider/base_controller.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 # Controller used to provide the API products for the DFC application
 module DfcProvider
-  class BaseController < ActionController::Base
+  class BaseController < ApplicationController
```
Comment (mkllnk) [inline engines/dfc_provider/app/controllers/dfc_provider/base_controller.rb:5]: No, this is the base for API controllers and they shouldn't have all the bloat of the ApplicationController. I think that the better solution is to rename this BaseController to ApplicationController within this module here. That should satisfy Rubocop.

**Review by mkllnk — CHANGES_REQUESTED** [review]
Thank you. Two things need to change here. I would drop the first two commits from this PR for quick merging and then start a new one.

**Comment by macanudo527** [comment]
@mkllnk OK, I took out the override and changed the name of the BaseController.

**Review by mkllnk — APPROVED** [review]

File: engines/dfc_provider/app/controllers/dfc_provider/application_controller.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 # Controller used to provide the API products for the DFC application
 module DfcProvider
-  class BaseController < ActionController::Base
+  class ApplicationController < ActionController::Base
```
Comment (dacook) [inline engines/dfc_provider/app/controllers/dfc_provider/application_controller.rb:5]: Took me a while to understand what was happening here. The [rule](https://www.rubydoc.info/gems/rubocop-rails/RuboCop/Cop/Rails/ApplicationController) says:

> Checks that controllers subclass ‘ApplicationController`.

But I think what the rule should  _really_ say is:

_Checks that the application uses a class named `ApplicationController`, and it is the only class that directly inherits from `ActionController::Base`_

**Review by dacook — APPROVED** [review]
👍

**Comment by dacook** [comment]
Although this are not "safe" autocorrections, I struggle to see how they cannot be 100% functionally equivalent. I think passing specs is enough testing here, and I'll take Maikel's approval as implicit agreement.

**Pull request merged**